eslint.config.mjs

```diff
@@ -412,6 +418,20 @@ export default defineConfig(
       ],
     },
   })),
+  {
+    files: [
+      `test/utils/**/*${EXTENSION_TS}`,
+      `docs/src/**/*${EXTENSION_TS}`,
+      `docs/data/**/*${EXTENSION_TS}`,
+    ],
+    ignores: ['**/*.d.ts', '**/*.spec{.ts,.tsx}', '**/*.test{.ts,.tsx}'],
+    rules: {
+      'react-hooks/immutability': 'off',
+      'react-hooks/globals': 'off',
+      'react-hooks/refs': 'off',
+      'react-hooks/static-components': 'off',
+    },
```

Comment:
Doesn't the [previous](https://github.com/mui/mui-x/pull/20128/files#diff-9601a8f6c734c2001be34a2361f76946d19a39a709b5e8c624a2a5a0aade05f2R182) (line 182) config turn this off for all files ?